### PR TITLE
Allow to disable ErrorProne

### PR DESCRIPTION
**Pull request by hmijail**
## PR Description
Simplifies javac calls to allow the build to be run by other analyzers.

Example:
```
./gradlew installDist -Davt.disableErrorProne=true
```
## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by ajsutton** [comment]
Do we really need to merge this?  Don't you need to modify the build to add other analysers anyway and so could just disable error prone as part of those uncommitted changes?

I'm not against merging it if there's significant value but would definitely prefer not to be able to disable error prone at all.

**Comment by hmijail** [comment]
No, I don't need to modify the build - apart from this change. 

With this change in, Infer (in this case) is run like 
```
infer run -- ./gradlew installDist -Davt.disableErrorProne=true
```

File: build.gradle [hunk 1]
```diff
@@ -125,6 +125,9 @@ allprojects {
     ]
 
     options.errorprone {
+      if ('true'.equalsIgnoreCase(System.getProperty('avt.disableErrorProne'))) {
+        enabled = false
+      }
```
Comment (ajsutton) [inline build.gradle]: Can we do this as just:
```suggestion
      enabled = !'true'.equalsIgnoreCase(System.getProperty('avt.disableErrorProne'))
```
?

**Review by ajsutton — APPROVED** [review]
LGTM.

**Pull request merged**